hs_file_types/utils.py

```diff
@@ -307,54 +332,119 @@ def ingest_logical_file_metadata(metadata_file, resource, map_files):
     if not agg_type_name:
         raise Exception("Could not derive aggregation type from {}".format(metadata_file.name))
     subject = None
-    for s, _, _ in graph.triples((None, DC.title, None)):
+    for s, _, _ in graph.triples((None, DC.type, None)):
         subject = s.split('/resource/', 1)[1].split("#")[0]
         break
     if not subject:
         raise Exception("Could not derive aggregation path from {}".format(metadata_file.name))
 
     logical_file_class = get_logical_file(agg_type_name)
     lf = get_logical_file_by_map_file_path(resource, logical_file_class, subject)
-
+    logical_files = {}
     if not lf:
         # see if the files exist and create it
-        res_file = None
-        if logical_file_class is FileSetLogicalFile:
+        res_files = []
+        folder_based_aggr_created = False
+        if logical_file_class is FileSetLogicalFile or logical_file_class is ModelInstanceLogicalFile or \
```

Comment:
These changes are reading as if model instance/program can be applied to a set of files or a folder... I'm not sure why it should matter, a folder resolves to a set of files.